packages/strapi/lib/core-api/queries/bookshelf.js

```diff
@@ -71,25 +71,25 @@ module.exports = ({ model, modelKey }) => {
     },
 
     async delete(params) {
-      params.values = {};
+      const values = {};
       model.associations.map(association => {
         switch (association.nature) {
           case 'oneWay':
           case 'oneToOne':
           case 'manyToOne':
           case 'oneToManyMorph':
-            params.values[association.alias] = null;
+            values[association.alias] = null;
             break;
           case 'oneToMany':
           case 'manyToMany':
           case 'manyToManyMorph':
-            params.values[association.alias] = [];
+            values[association.alias] = [];
             break;
           default:
         }
       });
 
-      await model.updateRelations(params);
+      await model.updateRelations({ ...params, values });
       return model.forge(params).destroy();
```

Comment:
```suggestion
      return model.forge(_.pick(params, model.primaryKey)).destroy();
```
Isn't that better?
Because there might be another property in params, but in here we only need the id, or otherwise it throws an error, right?

Reply:
@sepehrHosseini the default DELETE endpoint uses the ID, but this function is a general function for all queries. You can delete using any set of filters, not just IDs.